### PR TITLE
grub2: move grub-bios-setup and grub-editenv to none variant

### PR DESCRIPTION
**Pull request by uxgood**
Refactored the compilation of grub2:

    1.make grub2-bios-setup as a separate package
    
    The grub2 and grub2-efi packages should only contain boot-related code.
    grub-bios-setup is the same as grub-editenv, they are both grub2 tools
    and should be placed in a separate package.

    2.make grub2 tools built in a separate variant
    
    grub2 boot-related code and tools-related code may require different
    compilation parameters. We split them into different variants for
    compilation, so that we can accurately pass the required parameters and
    avoid causing problems.

    3.avoid generating empty grub2 and grub2-efi packages
    
    The grub2 and grub2-efi packages only compile the boot-related code,
    they are finally installed in $(STAGING_DIR_IMAGE)/grub2/, so these two
    packages become empty, so we set PKG_TARGETS := bin to avoid generating
    these two package.

    4.pass compilation parameters more accurately
    
    In order for the grub2 boot-related code to compile normally, we have
    made many adjustments to the compilation parameters. These adjustments
    are not necessary for tools-related code. We apply these parameter
    adjustments only to the boot-related code.

    5.make compilation parallel
    
    grub2 does not have any parallel compilation problems,
    we make it parallel compilation.

Compile tested: x86_64
Run tested: x86_64

### CODE REVIEW

**Comment by uxgood** [comment]
@ldir-EDB0

**Comment by ynezz** [comment]
I'm missing answer to `Why are the changes needed?` question in commit description. You're just describing what you've changed, but that's visible in the commit diff already.

**Comment by ynezz** [comment]
You're probably trying to do a lot of changes (7) in one single commit  and you should probably split this one huge commit into smaller ones with proper commit description.

**Comment by uxgood** [comment]
Updated. 
Split into 5 commits, and added more detailed commit log.

**Comment by uxgood** [comment]
Is it possible to be merged recently？

File: package/boot/grub2/Makefile [hunk 1]
```diff
@@ -18,10 +18,14 @@ PKG_SOURCE_URL:=@GNU/grub
 PKG_HASH:=e5292496995ad42dabe843a0192cf2a2c502e7ffcc7479398232b10a472df77d
 
 HOST_BUILD_PARALLEL:=1
+PKG_BUILD_PARALLEL:=1
```
Comment (aparcar) [inline package/boot/grub2/Makefile:21]: These two lines can be removed.

File: package/boot/grub2/Makefile [hunk 2]
```diff
@@ -23,6 +23,10 @@ PKG_BUILD_DEPENDS:=grub2/host
 PKG_ASLR_PIE:=0
 PKG_SSP:=0
 
+ifneq ($(BUILD_VARIANT),none)
+  PKG_TARGETS := bin
```
Comment (aparcar) [inline package/boot/grub2/Makefile:27]: While the intentions are good it removes the package from "opkg list" and therefore hides away the version installed on a running system. I'd keep it as is.

**Comment by aparcar** [comment]
Thanks I merged all commits except the obsolete parallel patch and the `bin` patch since I want to keep grub listed in `opkg list`.

**Pull request closed without merging**